File: integrations/aspire/src/Scalar.Aspire/Enums/CredentialsLocation.cs
using System.ComponentModel;
using NetEscapades.EnumGenerators;

namespace Scalar.Aspire;

/// <summary>
/// Specifies the location where authentication credentials should be placed in HTTP requests.
/// </summary>
[EnumExtensions]
public enum CredentialsLocation
{
    /// <summary>
    /// Credentials are included in the request body.
    /// </summary>
    [Description("body")]
    Body,

    /// <summary>
    /// Credentials are included in the request headers.
    /// </summary>
    [Description("header")]
    Header
}

File: integrations/java/webjar/src/main/java/com/scalar/maven/webjar/enums/CredentialsLocation.java
package com.scalar.maven.webjar.enums;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;

/**
 * Represents where OAuth2 credentials should be sent in HTTP requests.
 */
public enum CredentialsLocation {
    /**
     * Send credentials in the HTTP header.
     */
    HEADER("header"),

    /**
     * Send credentials in the HTTP body.
     */
    BODY("body");

    private final String value;

    CredentialsLocation(String value) {
        this.value = value;
    }

    /**
     * Creates a CredentialsLocation from a string value.
     *
     * @param value the string value
     * @return the corresponding CredentialsLocation
     * @throws IllegalArgumentException if the value is not recognized
     */
    @JsonCreator
    public static CredentialsLocation fromValue(String value) {
        for (CredentialsLocation location : values()) {
            if (location.value.equals(value)) {
                return location;
            }
        }
        throw new IllegalArgumentException("Unknown credentials location: " + value);
    }

    /**
     * Gets the string value for JSON serialization.
     *
     * @return the string value
     */
    @JsonValue
    public String getValue() {
        return value;
    }
}
